### DISCUSSION TITLE
Economy

### DISCUSSION THREAD

**Discussion by JackSteel97** [JackSteel97/SteelBot#19] [Ideas]
Initial design questions for built-in economy

- How do users earn "gold"
- How can users spend "gold"

Feature suggestions from users:

- Feeding pets grants temporary bonuses?
- Users can attempt to steal from other users
- A steal can critically fail or succeed, e.g. critical failure thief loses some of their own "gold"
- Buy/Sell accessories for pets
- Pet accessories provide you/your pets with bonuses

**Comment by LiamWilkinson98** [comment]
pay coins to reroll a stat slot? like reroll a deafen bonus into -1 pet slot or something random

**Comment by NoremacSkeew** [comment]
Pay to hinder other users like temp debuff

**Reply by JackSteel97** [reply]
Need to come up with some debuff types if there's any different ones to the negative of existing pet bonuses

**Comment by NoremacSkeew** [comment]
convert xp to gold

**Reply by JackSteel97** [reply]
XP should never really go down, none of the maths expects XP to go backwards

**Comment by AdamLindsay98** [comment]
More a point for the future after gold is implemented but it would be cool if you could link it to the stocks game to invest gold 

![C37A2F89-ECE7-4636-86E4-7692FECDCA7B](https://user-images.githubusercontent.com/102915408/161979995-c7ebec22-8ea8-4c74-b446-a53281be00bb.jpeg)

**Reply by NoremacSkeew** [reply]
That would be insane

**Reply by JackSteel97** [reply]
Don't think the stocks stuff will make it into general release because of the API rate-limit

**Comment by AdamLindsay98** [comment]
For earning gold could be you get a set amount of gold each time you level up with that number going up each era you progress into. Should be fairly easy to give gold to everybody retroactively that way

**Reply by JackSteel97** [reply]
The bot doesn't know anything about eras that's just our usage of Rank Roles, but yes leveling up can grant gold

**Comment by JackSteel97** [comment]
- primary pet can be sent out to "scavenge" once every `x` hours
- pets have a scavenger rating generated at birth
- scavenging can return gold or items or nothing - better ability = higher chance of better results
